### PR TITLE
feat(rest): Add POST/PATCH license endpoints

### PR DESCRIPTION
**Pull request by GMishx**
## Description

Added new POST and PATCH endpoints to `/license` for creating and editing licenses from REST API.

The GET to `/license` is also edited. Now, the request without license name will list every license (paginated). Request to `/license/<shortname>` will list just the requested license but will also include the obligations associated to it.

### Changes

1. Added the POST and PATCH endpoints to `/license`.
2. New model `Obligation`.
3. Added `/license` endpoint for listing all licenses.
4. Update `/license/<shortname>` endpoint to list license obligations.

## How to test

1. Try to get all the licenses from the `/license` endpoint and check if they matched the DB.
2. Check the pagination feature of the `/license` endpoint.
3. Check the obligation list of a license with `/license/<shortname>` endpoint.
4. Try creating a new license with `POST` request.
5. Try updating a license with `PATCH` request.

### CODE REVIEW

**Comment by ag4ums** [comment]
user is able to add multiple duplicate license with same shortname

File: src/www/ui/api/Controllers/LicenseController.php [hunk 1]
```diff
@@ -84,14 +104,227 @@ public function getLicense($request, $response, $args)
...
+    ];
+    if ($newLicense->getIsCandidate()) {
+      $tableName = "license_candidate";
+      $assocData["group_fk"] = $this->restHelper->getGroupId();
+      $assocData["rf_user_fk_created"] = $this->restHelper->getUserId();
+      $assocData["rf_user_fk_modified"] = $this->restHelper->getUserId();
+      $assocData["marydone"] = $newLicense->getMergeRequest();
+    }
+    $rfPk = $this->dbHelper->getDbManager()->insertTableRow($tableName,
+      $assocData, __METHOD__ . ".newLicense", "rf_pk");
```
Comment (ag4ums) [inline src/www/ui/api/Controllers/LicenseController.php:255]: duplicate candidate license check require before adding a new candidate license.

Comment (GMishx) [reply src/www/ui/api/Controllers/LicenseController.php]: @ag4ums I have added checks for duplicate shortname for both main license and candidate license. Also added a check for duplicate license text for main license.

**Review by ag4ums — APPROVED** [review]
Code looks good working as expected

**Pull request merged**